dGame/dComponents/TriggerComponent.cpp

```diff
@@ -0,0 +1,214 @@
+		case eTriggerCommandType::TURN_AROUND_ON_PATH: break;
+		case eTriggerCommandType::GO_FORWARD_ON_PATH: break;
+		case eTriggerCommandType::GO_BACKWARD_ON_PATH: break;
+		case eTriggerCommandType::STOP_PATHING: break;
+		case eTriggerCommandType::START_PATHING: break;
+		case eTriggerCommandType::LOCK_OR_UNLOCK_CONTROLS: break;
+		case eTriggerCommandType::PLAY_EFFECT: break;
+		case eTriggerCommandType::STOP_EFFECT: break;
+		case eTriggerCommandType::ACTIVATE_MUSIC_CUE: break;
+		case eTriggerCommandType::DEACTIVATE_MUSIC_CUE: break;
+		case eTriggerCommandType::FLASH_MUSIC_CUE: break;
+		case eTriggerCommandType::SET_MUSIC_PARAMETER: break;
+		case eTriggerCommandType::PLAY_2D_AMBIENT_SOUND: break;
+		case eTriggerCommandType::STOP_2D_AMBIENT_SOUND: break;
+		case eTriggerCommandType::PLAY_3D_AMBIENT_SOUND: break;
+		case eTriggerCommandType::STOP_3D_AMBIENT_SOUND: break;
+		case eTriggerCommandType::ACTIVATE_MIXER_PROGRAM: break;
+		case eTriggerCommandType::DEACTIVATE_MIXER_PROGRAM: break;
+		case eTriggerCommandType::CAST_SKILL: break;
+		case eTriggerCommandType::DISPLAY_ZONE_SUMMARY: break;
+		case eTriggerCommandType::SET_PHYSICS_VOLUME_EFFECT:
+			HandleSetPhysicsVolume(targetEntity, argArray, command->target);
+			break;
+		case eTriggerCommandType::SET_PHYSICS_VOLUME_STATUS: break;
+		case eTriggerCommandType::SET_MODEL_TO_BUILD: break;
+		case eTriggerCommandType::SPAWN_MODEL_BRICKS: break;
+		case eTriggerCommandType::ACTIVATE_SPAWNER_NETWORK: break;
+		case eTriggerCommandType::DEACTIVATE_SPAWNER_NETWORK: break;
+		case eTriggerCommandType::RESET_SPAWNER_NETWORK: break;
+		case eTriggerCommandType::DESTROY_SPAWNER_NETWORK_OBJECTS: break;
+		case eTriggerCommandType::GO_TO_WAYPOINT: break;
+		case eTriggerCommandType::ACTIVATE_PHYSICS: break;
+		}
+	}
+}
+
+std::vector<Entity*> TriggerComponent::GatherTargets(LUTriggers::Command* command, Entity* optionalTarget) {
+	std::vector<Entity*> entities = {};
+
+	if (command->target == "self") entities.push_back(m_Parent);
+	else if (command->target == "zone") { /*TODO*/ }
+	else if (command->target == "target") { /*TODO*/ }
+	else if (command->target == "targetTeam") { /*TODO*/ }
+	else if (command->target == "objGroup") entities = EntityManager::Instance()->GetEntitiesInGroup(command->targetName);
+	else if (command->target == "allPlayers") { /*TODO*/ }
+	else if (command->target == "allNPCs") { /*TODO*/ }
+
+	if (optionalTarget) entities.push_back(optionalTarget);
+
+	return entities;
+}
+
+std::vector<std::string> TriggerComponent::ParseArgs(std::string args){
+	std::vector<std::string> argArray;
+	std::stringstream ssData(args);
+	std::string token;
+	char deliminator = ',';
+
+	while (std::getline(ssData, token, deliminator)) {
+		std::string lowerToken;
+		for (char character : token) {
+			lowerToken.push_back(std::tolower(character)); // make lowercase to ensure it works
+		}
+		argArray.push_back(lowerToken);
+	}
+	return argArray;
+}
```

Comment:
Like in the constructor, could just replace all of this with `GeneralUtils::SplitString();`.  This code probably wasn't touched for a long time...